build.gradle

```diff
@@ -209,33 +272,61 @@ File getClientJar() {
             .get()
 }
 
-task copyMcJar(type: Copy) {
-    def mcJar = {getClientJar()}
-
-    from mcJar
-    into 'build/createMcSrgJar/'
-    rename {'client-srg.jar'}
+File getClientJarFabric() {
+    return project.getConvention().getPlugin(JavaPluginConvention.class).getSourceSets().findByName("launch").getRuntimeClasspath().getFiles()
+        .stream()
+        .filter({f -> f.getName().endsWith("-v2.jar") && f.getName().startsWith("minecraft-")})
+        .map({f -> new File(f.getParentFile().getParentFile(), f.getName().toString().replace("mapped", "intermediary"))})
+        .findFirst()
+        .get()
 }
 
-task createSrgMc(type: RenameJarInPlace) {
-    setInput(new File(copyMcJar.getOutputs().getFiles().getSingleFile(), "client-srg.jar"))
-    setClasspath(files({getClientJar()}))
+if (getProject().hasProperty("baritone.fabric_build")) {
+    task copyMcJar(type: Copy) {
+        def mcJar = {getClientJarFabric()}
 
-    // fork
-    setMappingType(net.minecraftforge.gradle.common.util.MappingFile.Mapping.SEARGE)
-    setJarTask('trans alaska pipeline')
-}
-project.afterEvaluate {
-    createSrgMc.dependsOn(extractSrg, copyMcJar)
-    createSrgMc.setMappings(extractSrg.getOutput())
-}
+        from mcJar
+        into 'build/createMcIntermediaryJar/'
+        rename {'client.jar'}
+    }
+
+    task proguard(type: ProguardTask, dependsOn: copyMcJar) {
+        url 'https://downloads.sourceforge.net/project/proguard/proguard/6.0/proguard6.0.3.zip'
+        extract 'proguard6.0.3/lib/proguard.jar'
+    }
 
+    task createDist(type: CreateDistTask, dependsOn: proguard)
 
-task proguard(type: ProguardTask, dependsOn: createSrgMc) { // TODO: dont need to create srg mc if doing notch build
-    url 'https://downloads.sourceforge.net/project/proguard/proguard/6.0/proguard6.0.3.zip'
-    extract 'proguard6.0.3/lib/proguard.jar'
-}
+} else {
+    task copyMcJar(type: Copy) {
+        def mcJar = {getClientJar()}
+
+        from mcJar
+        into 'build/createMcSrgJar/'
+        rename {'client-srg.jar'}
+    }
 
-task createDist(type: CreateDistTask, dependsOn: proguard)
+    task createSrgMc(type: RenameJarInPlace) {
+        setInput(new File(copyMcJar.getOutputs().getFiles().getSingleFile(), "client-srg.jar"))
+        setClasspath(files({getClientJar()}))
 
+        // fork
+        setMappingType(net.minecraftforge.gradle.common.util.MappingFile.Mapping.SEARGE)
+        setJarTask('trans alaska pipeline')
```

Comment:
is this a reference to my fucking reddit post LOL

Reply:
holy shit my inside joke

Reply:
I didn't touch any of the forge stuff I just tabbed it in as is xd.